src/main/java/com/commercetools/sync/sdk2/cartdiscounts/utils/CartDiscountReferenceResolutionUtils.java

```diff
@@ -0,0 +1,178 @@
+      @Nonnull final ReferenceIdToKeyCache referenceIdToKeyCache) {
+    return CartDiscountDraftBuilder.of()
+        .cartPredicate(cartDiscount.getCartPredicate())
+        .name(cartDiscount.getName())
+        .requiresDiscountCode(cartDiscount.getRequiresDiscountCode())
+        .sortOrder(cartDiscount.getSortOrder())
+        .target(cartDiscount.getTarget())
+        .value(toCartDiscountValueDraft(cartDiscount.getValue()))
+        .key(cartDiscount.getKey())
+        .description(cartDiscount.getDescription())
+        .isActive(cartDiscount.getIsActive())
+        .validFrom(cartDiscount.getValidFrom())
+        .validUntil(cartDiscount.getValidUntil())
+        .stackingMode(cartDiscount.getStackingMode())
+        .custom(mapToCustomFieldsDraft(cartDiscount, referenceIdToKeyCache))
+        .build();
+  }
+
+  private static CartDiscountValueDraft toCartDiscountValueDraft(
+      final CartDiscountValue cartDiscountValue) {
+    if (cartDiscountValue instanceof CartDiscountValueAbsolute) {
+      return cloneCartDiscountValueAbsoluteDraft((CartDiscountValueAbsolute) cartDiscountValue);
+    }
+    if (cartDiscountValue instanceof CartDiscountValueFixed) {
+      return cloneCartDiscountValueFixedDraft((CartDiscountValueFixed) cartDiscountValue);
+    }
+    if (cartDiscountValue instanceof CartDiscountValueGiftLineItem) {
+      return cloneCartDiscountValueGiftLineItemDraft(
+          (CartDiscountValueGiftLineItem) cartDiscountValue);
+    }
+    if (cartDiscountValue instanceof CartDiscountValueRelative) {
+      return cloneCartDiscountValueRelativeDraft((CartDiscountValueRelative) cartDiscountValue);
+    }
+    return null;
+  }
+
+  private static CartDiscountValueRelativeDraft cloneCartDiscountValueRelativeDraft(
+      final CartDiscountValueRelative cartDiscountValue) {
+    final CartDiscountValueRelative cartDiscountValueRelative = cartDiscountValue;
+    return CartDiscountValueRelativeDraftBuilder.of()
+        .permyriad(cartDiscountValueRelative.getPermyriad())
+        .build();
+  }
+
+  private static CartDiscountValueGiftLineItemDraft cloneCartDiscountValueGiftLineItemDraft(
+      final CartDiscountValueGiftLineItem cartDiscountValue) {
+    final CartDiscountValueGiftLineItem cartDiscountValueGiftLineItem = cartDiscountValue;
+    return CartDiscountValueGiftLineItemDraftBuilder.of()
+        .distributionChannel(
+            channelResourceIdentifierBuilder ->
+                channelResourceIdentifierBuilder.id(
+                    cartDiscountValueGiftLineItem.getDistributionChannel().getId()))
+        .product(
+            productResourceIdentifierBuilder ->
+                productResourceIdentifierBuilder.id(
+                    cartDiscountValueGiftLineItem.getProduct().getId()))
+        .supplyChannel(
+            channelResourceIdentifierBuilder ->
+                channelResourceIdentifierBuilder.id(
+                    cartDiscountValueGiftLineItem.getSupplyChannel().getId()))
```

Comment:
`getSupplyChannel()` might be null.

Reply:
https://github.com/commercetools/commercetools-sync-java/pull/1010/commits/1c3670042cc994fd23fe20e4cbea811ab2fa26a1